packages/expo/android/src/main/java/expo/modules/ApplicationLifecycleDispatcher.kt

```diff
@@ -2,24 +2,28 @@ package expo.modules
 
 import android.app.Application
 import android.content.res.Configuration
-import expo.modules.core.ModulePriorities
+import androidx.annotation.UiThread
+import expo.modules.core.interfaces.ApplicationLifecycleListener
 
-class ApplicationLifecycleDispatcher private constructor() {
-  companion object {
-    @JvmStatic
-    fun onApplicationCreate(application: Application) {
-      ExpoModulesPackage.packageList
-        .sortedByDescending { ModulePriorities.get(it::class.qualifiedName) }
-        .flatMap { it.createApplicationLifecycleListeners(application) }
-        .forEach { it.onCreate(application) }
-    }
+object ApplicationLifecycleDispatcher {
+  private var listeners: List<ApplicationLifecycleListener>? = null
```

Comment:
Maybe we can convert this property to be lazily initialised (`by lazy`)? That would enable us to get rid of this `getCachedListeners` method 🤔

Reply:
can kotlin `by lazy` pass an `application` parameter into the getter block?

Reply:
Valid question and I guess it's not possible directly as we're not having `application` variable in the class scope 🤔 
cc @lukmccall

Reply:
It's not possible directly and we could do a workaround, but I don't think it's worth it here.

Reply:
thanks @lukmccall.

@bbarthec so how do you think if we keep the original code? imo that is most straightforward one. 

e.g. 
```kotlin
    @JvmStatic
    fun onApplicationCreate(application: Application) {
      ExpoModulesPackage.packageList
        .flatMap { it.createApplicationLifecycleListeners(application) }
        .forEach { it.onCreate(application) }
    }
    
   @JvmStatic
    fun onConfigurationChanged(application: Application) {
      ExpoModulesPackage.packageList
        .flatMap { it.createApplicationLifecycleListeners(application) }
        .forEach { it. onConfigurationChanged(application) }
    }
```

Reply:
I think we will change how we obtain lifecycle listeners when I add support for them to the sweet API. So I don't think we should bother about the current version as long it works 😉

Reply:
Ok, I'm convinced then 👍 
I'm fine with the current version:
```kt
  @UiThread
  private fun getCachedListeners(application: Application): List<ApplicationLifecycleListener> {
    if (listeners == null) {
      listeners = ExpoModulesPackage.packagePrioritySortedList
        .flatMap { it.createApplicationLifecycleListeners(application) }
    }
    return listeners!!

  @JvmStatic
  fun onApplicationCreate(application: Application) {
    getCachedListeners(application).forEach { it.onCreate(application) }
  }

  @JvmStatic
  fun onConfigurationChanged(application: Application, newConfig: Configuration) {
    getCachedListeners(application).forEach { it.onConfigurationChanged(newConfig) }
  }
```